.github/workflows/notebook-test.yml

```diff
@@ -0,0 +1,67 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+name: Test notebooks
+on:
+  pull_request:
+    paths:
+      - "docs/**/*.ipynb"
+      - "!docs/api/**/*"
+  workflow_dispatch:
+jobs:
+  execute:
+    name: Execute notebooks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: "pip"
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@af2816c65436325c50621100d67f6e853cd1b0f1
+
+      - name: Install LaTeX dependencies
+        run: |
+          # Add your notebook to this list if it needs latex to run
+          NEEDS_LATEX="docs/build/circuit-visualization.ipynb"
+          for FILE in $NEEDS_LATEX; do
+            if [[ "${{ steps.changed-files.outputs.all_changed_files }}" = *"$FILE"* ]]; then
+              sudo apt-get install texlive-pictures texlive-latex-extra poppler-utils
+              break
+            fi
+          done
+
+      - name: Install Qiskit IBM Runtime to save account
+        # To save account; this is re-installed during the "Run tox" step
+        run: pip install qiskit-ibm-runtime
+
+      - name: Save IBM Quantum account
+        if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
+        shell: python
+        run: |
+          from qiskit_ibm_runtime import QiskitRuntimeService
+          QiskitRuntimeService.save_account(
+              channel="ibm_quantum",
+              instance="ibm-q/open/main",
+              token="${{ secrets.IBM_QUANTUM_TEST_TOKEN }}",
+              set_as_default=True
+          )
+
+      - name: Run tox
+        uses: lsst-sqre/run-tox@97818256d9fa3c72d0c12f31660718adb495a1cb
```

Comment:
This is not a very popular action. While it looks legit, there is some risk using it, including supply chain security. Probably use Tox explicitly like we normally do? While more verbose, it's not very complex code, unlike changed-files.